Anytype/Sources/PresentationLayer/Modules/HomeWidgets/Widgets/Common/WidgetContainerWithEmptyState.swift

```diff
@@ -0,0 +1,28 @@
+import Foundation
+import SwiftUI
+
+struct WidgetContainerWithEmptyState<Content: View>: View {
+    
+    let showEmpty: Bool
+    let content: Content
+    
+    init(showEmpty: Bool, @ViewBuilder content: () -> Content) {
+        self.showEmpty = showEmpty
+        self.content = content()
+    }
+    
+    var body: some View {
+        ZStack {
+            VStack(spacing: 0) {
+                WidgetEmptyView(title: Loc.Widgets.Empty.title)
+                    .frame(height: showEmpty ? 72 : 0)
+                Spacer.fixedHeight(8)
+            }
+            .setZeroOpacity(!showEmpty)
+            .transition(.opacity)
+            
+            content
+                .setZeroOpacity(showEmpty)
+        }
```

Comment:
Why did you not use `if {} else {}` and use ZStack instead?

Reply:
Not needed. Removed. ZStack was used in old implementation, very often